engine/src/exec-filters.cpp

```diff
@@ -699,7 +699,9 @@ void MCFiltersEvalBinaryDecode(MCExecContext& ctxt, MCStringRef p_format, MCData
                     bool t_space_skipped = false;
                     uindex_t t_temp_size = t_size;
                     
-                    while (!t_space_skipped)
+                    // We need t_temp_size > 0 to prevent unsigned int underflow as well as an
+                    // array-out-of-bounds exception when it is 0`
+                    while (!t_space_skipped && t_temp_size > 0)
                     {
                         if (offset > offset + t_temp_size - t_space_length)
```

Comment:
@livecodepanos I think this line, rather than the above is where the fault is - this comparison will always fail when offset is 0, i.e. when reading from the beginning of the data. This can be rewritten to simplify as `if (t_space_length > t_temp_size)` which makes sense as you want to stop looking for spaces when an encoded space char won't fit within the remaining data.

Reply:
Thanks @livecodeian, I have updated the PR